qa/rpc-tests/sc_proving_system_selection.py

```diff
@@ -0,0 +1,488 @@
+        assert_equal(scinfo0['cswProvingSystem'], csw_proving_system)
+
+        mark_logs("Node0 generating 1 block", self.nodes, DEBUG_MODE)
+        prev_epoch_block_hash = self.nodes[0].getblockhash(self.nodes[0].getblockcount())
+        blocks.extend(self.nodes[0].generate(1))
+        self.sync_all()
+
+        mark_logs("Verify that sidechain configuration is as expected after connecting the new block", self.nodes, DEBUG_MODE)
+        scinfo0 = self.nodes[0].getscinfo(scid)['items'][0]
+        assert_equal(scinfo0['certificateProvingSystem'], cert_proving_system)
+        assert_equal(scinfo0['cswProvingSystem'], csw_proving_system)
+
+
+        # ---------------------------------------------------------------------------------------
+        # Node 0 - Create a valid sidechain with createrawtransaction()
+        mark_logs("\nNode 0 creates a new sidechain with createrawtransaction()", self.nodes, DEBUG_MODE)
+
+        sc_cr = [{
+            "epoch_length": withdrawalEpochLength, "amount": creation_amount, "address": address,
+            "certProvingSystem": proving_system, "wCertVk": vk,
+            "wCeasedVk": cswVk, "cswProvingSystem": csw_proving_system }]
+
+        try:
+            rawtx = self.nodes[0].createrawtransaction([],{},[],sc_cr)
+            funded_tx = self.nodes[0].fundrawtransaction(rawtx)
+            sigRawtx = self.nodes[0].signrawtransaction(funded_tx['hex'])
+            creating_tx = self.nodes[0].sendrawtransaction(sigRawtx['hex'])
+        except JSONRPCException, e:
+            errorString = e.error['message']
+            mark_logs(errorString,self.nodes,DEBUG_MODE)
+            assert_true(False)
+
+        self.sync_all()
+
+        mark_logs("Verify that the proving systems have been set correctly on the transaction JSON object", self.nodes, DEBUG_MODE)
+        decoded_tx = self.nodes[0].getrawtransaction(creating_tx, 1)
+        assert_equal(cert_proving_system, decoded_tx['vsc_ccout'][0]['certProvingSystem'])
+        assert_equal(csw_proving_system, decoded_tx['vsc_ccout'][0]['cswProvingSystem'])
+
+        mark_logs("Verify that sidechain configuration is as expected [mempool]", self.nodes, DEBUG_MODE)
+        scinfo0 = self.nodes[0].getscinfo(scid)['items'][0]
+        assert_equal(scinfo0['certificateProvingSystem'], cert_proving_system)
+        assert_equal(scinfo0['cswProvingSystem'], csw_proving_system)
+
+        mark_logs("Node0 generating 1 block", self.nodes, DEBUG_MODE)
+        prev_epoch_block_hash = self.nodes[0].getblockhash(self.nodes[0].getblockcount())
+        blocks.extend(self.nodes[0].generate(1))
+        self.sync_all()
+
+        mark_logs("Verify that sidechain configuration is as expected after connecting the new block", self.nodes, DEBUG_MODE)
+        scinfo0 = self.nodes[0].getscinfo(scid)['items'][0]
+        assert_equal(scinfo0['certificateProvingSystem'], cert_proving_system)
+        assert_equal(scinfo0['cswProvingSystem'], csw_proving_system)
+
```

Comment:
Consider adding a stop/restart phase for checking persistance of the configuration